### DISCUSSION TITLE
Quale Package Manager dovremmo utilizzare?

### DISCUSSION THREAD

**Discussion by Byloth** [vuejsita/vuejsita.github.io#32] [General]
Arriva tutto dalla PR #31...

Il progetto -semplicemente per mia pura comodità- è nato con l'utilizzo di `yarn` ma @sariodesign -nella sua PR- ha utilizzato `npm` (corretto?).  
Quale standard -quindi- dovremmo adottare?

- [`npm`](https://nodejs.org/en/learn/getting-started/an-introduction-to-the-npm-package-manager) permetterebbe a chiunque -anche i meno esperti del Mondo Node.js- di entrare rapidamente nel progetto.
- [`yarn`](https://yarnpkg.com/) a dirla tutta -ora come ora e alla versione che stiamo usando- non presenta particolari benefit su `npm`.

Potremmo -però- adottare anche noi [`pnpm`](https://pnpm.io/); ho notato che -già da un po'- i veri PRO si stanno spostando tutti in questa direzione... 🤔

Cosa ne pensate?

**Comment by mitto98** [comment]
Io sto gradualmente spostando tutti i miei progetti su pnpm e mi sto trovando oggettivamente molto bene. Lo avevo provato per ridurre la quantità di file scritti su disco (grazie mamma Apple che li salda sulla mobo :/), ma mi ha piacevolmente stupito per la sua velocità e per l'approccio che ha ai monorepo.

Yarn ha fatto ormai la sua storia e Npm non ha più nulla da invidiargli

**Comment by longo-andrea** [comment]
Io è da diversi anni che uso pnpm e sono più che soddisfatto, in realtà lo vedo quasi meno 'da PRO' rispetto a yarn o package manger più recenti, nel senso che a livello di cli supporta i comandi nel formato di tutti i package manger più famosi, quindi la migrazione non è spaesante per nessuno.
Un suo lato super positivo, per me almeno, è il fatto di essere costruito sopra a npm, non si è inventato nulla di nuovo sfrutta tutto quello che già funziona e ci fa semplicemente ottimizzazioni.
Lato negativo, se sei incline a fare tanti typo come me, pnpm lo sbagli a scrivere 2 su 3 😂

**Reply by mitto98** [reply]
Io ho risolto i tuoi continui facendomi un alias a pn 😂

**Comment by Byloth** [comment]
Mi verrebbe quasi da dire che la "battaglia" si gioca tra `npm` e `pnpm`...  
Secondo voi, cos'ha più senso fare?

Fornire una curva d'ingresso al progetto morbida usando uno strumento che -bene o male- sanno usare tutti (`npm`) oppure andare sull'ultima tecnologia "cutting-edge" per dare l'esempio a tutta la community (`pnpm`)? 🤔

**Reply by mitto98** [reply]
Essendo pnpm un drop in replacement a npm non dovrebbe stravolgere nessuno. Comunque non è una scelta che deve rimanere scolpita nella pietra. Nulla vieta un cambio nel momento in cui esce qualcosa di migliore o npm ritorna in auge

**Comment by LuigiMigliardi** [comment]
Io uso yarn da anni e mi trovo benissimo

**Comment by longo-andrea** [comment]
Quindi, come procediamo su questo tema? Se non siamo tutti convintissimi di pnpm possiamo anche procedere on yarn visto che il progetto è inizializzato cosi

**Comment by Byloth** [comment]
Ho appena aperto la PR #36.  
Fateci una review, se -per voi- va bene o meno... 😉

**Reply by sariodesign** [reply]
Per me va bene pnpm
